src/pevm.rs

```diff
@@ -231,99 +231,136 @@ impl Pevm {
 
         // We fully evaluate (the balance and nonce of) the beneficiary account
         // and raw transfer recipients that may have been atomically updated.
-        for address in mv_memory.consume_lazy_addresses() {
-            let location_hash = self.hasher.hash_one(MemoryLocation::Basic(address));
-            if let Some(write_history) = mv_memory.data.get(&location_hash) {
-                let mut balance = U256::ZERO;
-                let mut nonce = 0;
-                // Read from storage if the first multi-version entry is not an absolute value.
-                if !matches!(
-                    write_history.first_key_value(),
-                    Some((_, MemoryEntry::Data(_, MemoryValue::Basic(_))))
-                ) {
-                    if let Ok(Some(account)) = storage.basic(&address) {
-                        balance = account.balance;
-                        nonce = account.nonce;
+        for memory_location in mv_memory.consume_lazy_locations() {
+            let location_hash = self.hasher.hash_one(memory_location.clone());
+            let Some(write_history) = mv_memory.data.get(&location_hash) else {
```

Comment:
This is good but the change is too noisy for this non-trivial work. I'd suggest doing it in another PR.

Reply:
I reverted the change. However, the added indentation is inevitable.